src/xWELL/xWELL.sol

```diff
@@ -0,0 +1,299 @@
+    }
+
+    /// --------------------------------------------------------
+    /// --------------------------------------------------------
+    /// -------------------- clock override --------------------
+    /// --------------------------------------------------------
+    /// --------------------------------------------------------
+
+    /// @notice override the clock in ERC20 Votes to use block timestamp
+    /// now all checkpoints use unix timestamp instead of block number
+    function clock() public view override returns (uint48) {
+        /// do not safe cast, overflow will not happen for billions of years
+        /// Given that the Unix Epoch started in 1970, adding these years to 1970 gives a theoretical year:
+        /// 1970 + 8,923,292,862.77 ≈ Year 8,923,292,883,832
+        return uint48(block.timestamp);
+    }
+
+    /// @dev Machine-readable description of the clock as specified in EIP-6372.
+    /// https://eips.ethereum.org/EIPS/eip-6372
+    // solhint-disable-next-line func-name-mixedcase
+    function CLOCK_MODE() public view override returns (string memory) {
+        // Check that the clock is correctly modified
+        require(clock() == uint48(block.timestamp), "Incorrect clock");
+
+        return "mode=timestamp";
+    }
+
+    ///  ------------------------------------------------------------
+    ///  ------------------------------------------------------------
+    ///  ------------------- Overridden Pure Hooks ------------------
+    ///  ------------------------------------------------------------
+    ///  ------------------------------------------------------------
+
+    /// @notice maximum supply is 5 billion tokens if all WELL holders migrate to xWELL
+    function maxSupply() public pure override returns (uint256) {
+        return MAX_SUPPLY;
+    }
+
+    /// @notice the maximum amount of time the token can be paused for
+    function maxPauseDuration() public pure override returns (uint256) {
+        return MAX_PAUSE_DURATION;
+    }
+
+    /// @notice the maximum rate limit per second
+    function maxRateLimitPerSecond() public pure override returns (uint128) {
+        return MAX_RATE_LIMIT_PER_SECOND;
+    }
+
+    function minBufferCap() public pure override returns (uint112) {
+        return MIN_BUFFER_CAP;
+    }
+
+    /// ------------------------------------------------------------
+    /// ------------------------------------------------------------
+    /// ------------------- Overridden View Hooks ------------------
+    /// ------------------------------------------------------------
+    /// ------------------------------------------------------------
+
+    /// @notice the total supply of the token
+    function totalSupply()
+        public
+        view
+        override(ERC20Upgradeable, xERC20)
+        returns (uint256)
+    {
+        return super.totalSupply();
```

Comment:
Since we're overriding two inherited contracts, it's worth clarifying which `super` contract is called.